lib/blocks.php

```diff
@@ -192,6 +194,8 @@ function gutenberg_register_core_block_styles( $block_name ) {
 			filemtime( gutenberg_dir_path() . $editor_style_path )
 		);
 		wp_style_add_data( "wp-block-{$block_name}-editor", 'rtl', 'replace' );
+	} else {
+		wp_register_style( "wp-block-{$block_name}-editor", false );
```

Comment:
The same question related to the case when the style handle is already registered.

Is it something we should add to WordPress core as well? I guess we still could.

Reply:
There are no conflicts with core.
Yes, this is something that should be added to core... Ideally this part would not be inside a condition and the `register_block_style_handle` function should be called regardless of whether the block has a `style` or `editorStyle` defined. This would mean that all the tweaks to the `block.json` files could be removed as well from this PR

Reply:
We will need this for backward compatibility reasons to work with older versions of WordPress. So I think we need both.

Reply:
I removed all tweaks to `block.json` files and instead used the `block_type_metadata` filter to add them where missing. I think this is a lot safer... :+1:

Reply:
`block_type_metadata` was added in WP 5.7.0:
https://github.com/WordPress/wordpress-develop/blob/59e55af520fd408d0389b683a073156164393ba6/src/wp-includes/blocks.php#L203-L210

Gutenberg still supports WP 5.6, but it will change in a few weeks. We can also try using `register_block_type_args` since we use a handle that is passed further without any modifications:

https://github.com/WordPress/wordpress-develop/blob/8add05ff150f148d3a5b9f88ce12c05d5cc48897/src/wp-includes/class-wp-block-type.php#L330-L338